src/web/handlers/middleware.rs

```diff
@@ -0,0 +1,32 @@
+use crate::service::ServiceContext;
+use rocket::http::Status;
+use rocket::request::{FromRequest, Outcome, Request};
+use rocket::State;
+
+pub struct IdentityCheck;
+
+#[derive(Debug)]
+pub struct IdentityCheckError;
+
+#[rocket::async_trait]
+impl<'r> FromRequest<'r> for IdentityCheck {
+    type Error = Status;
+
+    async fn from_request(request: &'r Request<'_>) -> Outcome<Self, Self::Error> {
+        let path = request.uri().path();
+
+        if path.starts_with("/identity") {
```

Comment:
Looks good, but this check would have to be added to every endpoint except for `/identity/create`. (It wasn't added everywhere, but we need it everywhere except for `/identity/create`.

So the IdentityCheck would have to be added to all endpoints in:
* Contacts
* Quotes
* Bill
* Company
* Identity (only add it to /identity/change)

Reply:
Yes, it wasn't consistently added before.

Perfect, thanks!